sahi/model.py

```diff
@@ -663,3 +665,156 @@ def _create_object_prediction_list_from_original_predictions(
         object_prediction_list_per_image = [object_prediction_list]
 
         self._object_prediction_list_per_image = object_prediction_list_per_image
+
+
+class TorchVisionDetectionModel(DetectionModel):
+    def load_model(self):
+        try:
+            import torchvision
+        except ImportError:
+            raise ImportError(
+                "torchvision is not installed. Please run 'pip install -U torchvision to use this "
+                "torchvision models'"
+            )
+
+        # set model
+        try:
+            from sahi.utils.torch import torch
+
+            model = self.config_path
+            model.load_state_dict(torch.load(self.model_path))
+            model.eval()
+            self.model = model.to(self.device)
+        except Exception as e:
+            TypeError("model_path is not a valid torchvision model path: ", e)
+
+        # set category_mapping
+        from sahi.utils.torchvision import COCO_CLASSES
+
+        if self.category_mapping is None:
+            category_names = {str(i): COCO_CLASSES[i] for i in range(len(COCO_CLASSES))}
+            self.category_mapping = category_names
+
+    def perform_inference(self, image: np.ndarray, image_size: int = None):
+        """
+        Prediction is performed using self.model and the prediction result is set to self._original_predictions.
+        Args:
+            image: np.ndarray
+                A numpy array that contains the image to be predicted. 3 channel image should be in RGB order.
+            image_size: int
+                Inference input size.
+        """
+        # TO DO: Check image size parameter(dont image_size parameter)
+        from sahi.utils.torchvision import numpy_to_torch, resize_aspect_ratio
+
+        if image_size is not None:
```

Comment:
Burada typo var, dogrusu su sekilde olmali:
```suggestion
        if self.image_size is not None:
```